libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -933,6 +933,25 @@ string YulUtilFunctions::calldataArrayIndexAccessFunction(ArrayType const& _type
 	});
 }
 
+string YulUtilFunctions::calldataArrayIndexRangeAccess(ArrayType const& _type)
+{
+	solAssert(_type.dataStoredIn(DataLocation::CallData), "");
+	string functionName = "calldata_array_index_range_access_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(offset, length, startIndex, endIndex) -> offsetOut, lengthOut {
+				if gt(startIndex, endIndex) { invalid() }
+				if gt(endIndex, length) { invalid() }
+				offsetOut := add(offset, mul(startIndex, <stride>))
+				lengthOut := sub(endIndex, startIndex)
+			}
+		)")
+			("functionName", functionName)
+			("stride", to_string(_type.calldataStride()))
+			.render();
```

Comment:
Looks like this is indented one tab too much... although it doesn't seem to be entirely consistent in the rest of the code either...

Reply:
Let's fix it than at least here :)

Reply:
By the way: in case you manage to come up with a CLion-coding-style-config that *exactly* matches the solidity coding style, let me know :-D. So far I only have a crude approximation myself :-).